### ISSUE TITLE
Deadlocks when debugging ThreadSendValues and ThreadReceiveValues

### ISSUE THREAD

**Issue by TanYuzhen** [encryptogroup/ABY#211]
Hello, author.
I'm debugging ABYcore and I'm running the `PerformInteraction()` function when two Parties interact. However, when I debug with gdb, I use the gdb command `set schedule-locking on` for the purpose of switching to the Client for receiving after the Server performs sending. Similarly, I want the Client to send and then the Server to receive.
Unfortunately, after I do this, the program deadlocks. Can you tell me why?
Because when I debug this part, if I don't execute this gdb command, only one of the two functions `ThreadSendValues()` and `ThreadReceiveValues()` can be debugged, and even if the other one adds a breakpoint in it, it is executed immediately, and I can't catch it.
In other words, I can't properly debug both functions `ThreadSendValues()` and` ThreadReceiveValues()` in a single participant, gdb is only able to catch one function and the other is executed automatically. Can you please tell me how I can solve this problem.
Looking forward to your reply!
![1726739099020](https://github.com/user-attachments/assets/a995c720-2247-45bd-8053-1196b56bd5e5)
